src/content/en/updates/posts/2016/03/web-enabled-internet-of-things.markdown

```diff
@@ -0,0 +1,523 @@
+
+You can find more info [here](https://github.com/don/node-eddystone-beacon/issues/30)
+
+The `package.json` file for the project looks like this:
+
+{% highlight javascript %}
+{
+  "name": "edison-webbluetooth-demo-server",
+  "version": "1.0.0",
+  "main": "main.js",
+  "engines": {
+    "node": ">=0.10.0"
+  },
+  "dependencies": {
+    "eddystone-beacon" : "1.0.5",
+    "parse-color": "1.0.0",
+    "johnny-five": "0.9.30"
+  }
+}
+{% endhighlight %}
+
+## Announcing the web site
+
+The beta version of Chrome on Android supports Physical Web, which
+allows Chrome to see URLs being broadcast by devices around it.
+There are a few requirements, such as the sites need to be publicly
+accessible and use HTTPS.
+
+My demo app is hosted at https://edison-webbt.appspot.com/ but the
+Eddystone protocol has a limit on the site of the URL (max 18 bytes,
+more info here), so I have to use a URL shortener in order for it to work.
+
+Broadcasting the URL is quite simple. Import the required libraries and
+call a few functions. One way of doing it is calling advertiseUrl, when
+the BLE chip is turned on:
+
+{% highlight javascript %}
+var beacon = require("eddystone-beacon");
+var bleno = require('eddystone-beacon/node_modules/bleno');
+
+bleno.on('stateChange', function(state) {    
+  if (state === 'poweredOn') {
+    beacon.advertiseUrl("https://goo.gl/9FomQC", {name: 'Edison'});
+  }   
+}
+{% endhighlight %}
+
+That really couldn't be much easier. Below you see that Chrome finds
+my device nicely.
+
+> NOTE: The Physical Web enabled devices only show up on your phone
+> when Bluetooth is turned on, and you launch Chrome (currently only
+> works with Beta). Additionally, you have to opt into the feature
+> first time you launch Chrome beta.
+
+<img alt="Chrome announces nearby Physical Web beacons" src="/web/updates/images/2016/03/web-enabled-internet-of-things/nearby.png" style="width: 50%; float: left"/>
+<img alt="The web app URL is listed" src="/web/updates/images/2016/03/web-enabled-internet-of-things/physicalweb.png" style="width: 50%; float: left"/>
+
+## Communicating with the sensor/actuator
+
+For talking to my board enhancements I used Johnny-Five. In simple
+cases like this, this is not strictly easier than talking manually,
+but for bigger projects it can be a real help. Johnny-Five has a nice
+abstraction for talking to a TMP36 sensor, but for some reason it only
+returned "undefined" as the current temperature so I went ahead and
+read the temperature value manually.
+
+Below you find the simple code for listening to temperature changes
+as well as setting the initial LED color.
+
+{% highlight javascript %}
+var five = require("johnny-five");
+var Edison = require("edison-io");
+var board = new five.Board({
+  io: new Edison()
+});
+
+board.on("ready", function() {
+  var led = new five.Led.RGB({
+      pins: {
+          red: 3,
+          green: 5,
+          blue: 6
+      },
+  });
+
+  colorCharacteristic._led = led;
```

Comment:
something off happens with github's diff here - everything goes italic? Just pointing it out in case it messes with something in the rendered output